Shared/Samples/Filter building scene layer/FilterBuildingSceneLayerView.swift

```diff
@@ -0,0 +1,208 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import ArcGISToolkit
+import SwiftUI
+
+struct FilterBuildingSceneLayerView: View {
+    /// A scene with the building scene layer which displays ESRI building E.
+    @State private var scene = Scene(url: URL(string: "https://www.arcgis.com/home/item.html?id=b7c387d599a84a50aafaece5ca139d44")!)!
+    
+    /// The building scene layer used in the scene.
+    ///
+    /// This is used to set the active filter on the layer.
+    @State private var buildingSceneLayer: BuildingSceneLayer?
+    
+    /// A Boolean value indicating if the settings are visible or not.
+    @State private var settingsAreVisible = false
+    
+    /// The available floors to show in the settings.
+    @State private var availableFloors: [String] = []
+    
+    /// The selected floor from the floor picker in the settings.
+    @State private var selectedFloor: String = .allFloorsLabel
```

Comment:
Instead of preloading this with "All", I suggest disabling the settings when there are no available floors. Then, `allFloorsLabel` would only be used in one place and you could inline it there.